generative_agents/modeling/langchain_agent.py

```diff
@@ -26,7 +25,7 @@ class GenerativeAgent(BaseModel):
     status: str
     """Current activities of the character."""
     llm: BaseLanguageModel
-    memory_retriever: TimeWeightedVectorStoreRetriever
+    memory_retriever: BaseRetriever
```

Comment:
At some point, should probably allow for both in the case that we don't use pinecone

Reply:
I agree completely!

ideal situation: langchain replaces the "raise NotImplementedError" in _similarity_search_with_relevance_scores() with an actual implementation

intermediate step: we can put in an "if not params["is_pinecone"]"